include/wx/osx/core/private/strconv_cf.h

```diff
@@ -316,9 +316,10 @@ class wxMBConv_cf : public wxMBConv
     {
     }
 
-    void Init( CFStringEncoding encoding)
+    void Init( CFStringEncoding encoding, bool compose_decompose = false)
```

Comment:
This is nitpicking, of course, but I'd strongly prefer having
```
enum NormalForm
{
    NF_Composed,
    NF_Decomposed
};
```
instead because `Init(..., true)` is really unreadable.

Reply:
I didn't want this to be in the public API, and it would have been for both composing and decomposing, so it would be a switch on and off, composing when reading in from utf8, decomposing when writing out to utf8. Then  I'd like to make this a bit field, having one bit setting composing when reading in and another when writing out, that way we can always decide

Reply:
So the whole range would become something like this

```
    enum NormalizationForm {
        None = 0x00,
        
        FromWChar_C = 0x02,
        FromWChar_D = 0x03,
        FromWChar_Mask = 0x03,
        
        ToWChar_C = 0x08,
        ToWChar_D = 0x0C,
        ToWChar_Mask = 0x0C
    };
```

so the default to preserve behavior would be `ToWChar_C`, for wxConvFilename this would be `ToWChar_C |  FromWChar_D`,

what do you think about that ? and do we have a template somewhere for bitwise operators on enums ?

Reply:
I don't mind but, to be honest, this looks like overkill to me as I don't think we actually need all the possible combinations, I'd just keep using NFC internally because this is the least surprising form and provide an option to use NFD for macOS file (and maybe others, I still don't know if it's also needed elsewhere) functions.

We don't have any template for enum bit operations and IMO it wouldn't be worth it to write it just for this enum.

Reply:
I'm ok with keeping the behavior NFC by default, but I'd like to make it explicit in the API, so I'll just keep 

    enum NormalizationForm {
        None = 0x00,
        FromWChar_C = 0x01,
        ToWChar_D = 0x02,
    };